lib/task_helper.rb

```diff
@@ -16,13 +16,35 @@ def run_local_command(command, wd = Dir.pwd)
   stdout
 end
 
+def platform_is_windows?(platform)
+  # TODO: This seems sub-optimal. We should be able to override/specify what the real platform is on a per target basis
+  # (plain_windows)            somewinorg/blah-windows-2019
+  # (plain_windows)            myorg/some_image:windows-server
+  # (bare_win_with_demlimiter) myorg/some_image:win-server-2008
+  # (bare_win_with_demlimiter) myorg/win-2k8r2
+  # No Match                   myorg/winderping    <--- Is this a Windows platform?
+  # (plain_windows)            myorg/windows-server
+  # (plain_windows)            windows-server
+  # (bare_win_with_demlimiter) win-2008
+  # (plain_windows)            webserserver-windows-2008
+  # (bare_win_with_demlimiter) webserver-win-2008
+  windows_regex = %r{(?<plain_windows>windows)|(?<bare_win_with_demlimiter>(?:^|[\/:\-\\;])win(?:[\/:\-\\;]|$))}
+  platform =~ windows_regex
+end
+
```

Comment:
do we need both of these ?

Reply:
Both of the regex matches? I think so, because there's a number of valid names either of them will miss.

Reply:
sorry my bad, i am not good at explaining. 
i realised the 2 windows detection functions are for different things. one to detect if the image is running, one to detect if it is running on windows

Reply:
am i correct ?

Reply:
Yes, whoops, that is correct. We need to fork behavior a little bit for both cases.